#!/usr/bin/env python
PACKAGE="open_simplex_noise_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("width", int_t, 1, "width pixels", 640, 1, 2048)
gen.add("height", int_t, 1, "height pixels", 480, 1, 2048)
gen.add("frame_rate", double_t, 1, "frame rate Hz", 20, 0.1, 50)
# TODO(lucasw) have 3 or 4 different independent controls,
# the results will be summed together
gen.add("feature_size_x", double_t, 2, "feature size x", 16.0, 0.1, 1024.0)
gen.add("feature_size_y", double_t, 2, "feature size y", 16.0, 0.1, 1024.0)
gen.add("feature_size_t", double_t, 2, "feature size t", 16.0, 0.1, 1024.0)
gen.add("scale", double_t, 2, "scale", 127.0, 0.1, 1024.0)
gen.add("offset", double_t, 2, "offset", 127.0, 0.0, 255.0)

exit(gen.generate(PACKAGE, "open_simplex_noise_ros", "OpenSimplexNoise"))
